test/integration.test.js

```diff
@@ -0,0 +1,439 @@
+'use strict';
+
+/* eslint-disable no-console, no-unused-vars, func-names, no-await-in-loop */
+const dns = require('dns');
+const exec = require('child_process').exec;
+const chai = require('chai');
+const request = require('request-promise-native');
+const aws = require('aws-sdk');
+
+const expect = chai.expect;
+
+const AWS_PROFILE = process.env.AWS_PROFILE;
+const TEST_DOMAIN = process.env.TEST_DOMAIN;
+
+function sleep(seconds) {
+  return new Promise(resolve => setTimeout(resolve, 1000 * seconds));
+}
+
+async function linkPackages() {
+  return new Promise((resolve, reject) => { // eslint-disable no-unused-vars
+    exec('npm link serverless-domain-manager', (err, stdout, stderr) => {
+      if (err || stderr) {
+        return resolve(false);
+      }
+      return resolve(true);
+    });
+  });
+}
+
+async function curlUrl(url) {
+  let response = null;
+  response = await request.get({
+    url,
+    resolveWithFullResponse: true,
+  })
+  .catch((err) => {
+    response = null;
+  });
+  if (response === undefined || response === null) {
+    return null;
+  }
+  return Promise.resolve(response.statusCode);
+}
+
+async function getEndpointType(url) {
+  let result = null;
+  const apigateway = new aws.APIGateway({
+    region: 'us-west-2',
+    credentials: new aws.SharedIniFileCredentials(
+      { profile: AWS_PROFILE },
+    ),
+  });
+  const params = {
+    domainName: url,
+  };
+  try {
+    result = await apigateway.getDomainName(params).promise();
+    return Promise.resolve(result.endpointConfiguration.types[0]);
+  } catch (err) {
+    return Promise.resolve(null);
+  }
+}
+
+async function getStage(url) {
+  let result = null;
+  const apigateway = new aws.APIGateway({
+    region: 'us-west-2',
+    credentials: new aws.SharedIniFileCredentials(
+      { profile: AWS_PROFILE },
+    ),
+  });
+  const params = {
+    domainName: url,
+  };
+  try {
+    result = await apigateway.getBasePathMappings(params).promise();
+    return Promise.resolve(result.items[0].stage);
+  } catch (err) {
+    return Promise.resolve(null);
+  }
+}
+
+async function getBasePath(url) {
+  let result = null;
+  const apigateway = new aws.APIGateway({
+    region: 'us-west-2',
+    credentials: new aws.SharedIniFileCredentials(
+      { profile: AWS_PROFILE },
+    ),
+  });
+  const params = {
+    domainName: url,
+  };
+  try {
+    result = await apigateway.getBasePathMappings(params).promise();
+    return Promise.resolve(result.items[0].basePath);
+  } catch (err) {
+    return Promise.resolve(null);
+  }
+}
+
+function deployLambdas(folderName) {
+  return new Promise((resolve, reject) => {
+    exec(`cd test/${folderName} && sls create_domain && sls deploy`, (err, stdout, stderr) => {
```

Comment:
Put `folderName` in single quotes so users don't get access to the machine or use a shell library.

Reply:
Use a shell library for all `exec` calls